core/src/tile/tileWorker.cpp

```diff
@@ -94,10 +97,10 @@ void TileWorker::run() {
 
             // float loadTime = (float(clock() - begin) / CLOCKS_PER_SEC) * 1000;
             // LOG("loadTime %s - %f", task->tile->getID().toString().c_str(), loadTime);
-
-            m_tileManager.tileProcessed(std::move(task));
```

Comment:
Hmm yes, the problem here is that the task must be passed back to TileManager to unset the 'reloading' flag of the original tile (by replacing it in tileset) otherwise future updates of the DataSource would not reload the tile.